### PR TITLE
Add thread-safe method to state store object

### PR DESCRIPTION
**Pull request by tomchop**
And use them in the Turbinia processor.

### CODE REVIEW

File: dftimewolf/lib/processors/turbinia.py [hunk 1]
```diff
@@ -102,26 +102,51 @@ def process(self):
     request = TurbiniaRequest()
     request.evidence.append(evidence_)
 
+    # Get threat intelligence data from any modules that have stored some.
+    observables = self.state.get_data('threat_intelligence')
+    if observables:
```
Comment (Onager) [inline dftimewolf/lib/processors/turbinia.py]: As discussed, adding a comment here as to what "observables" is supposed to look like would be nice. Perhaps also make an Observable class to make this more checkable

File: dftimewolf/lib/processors/turbinia.py [hunk 2]
```diff
@@ -102,26 +102,51 @@ def process(self):
...
       return
 
+    # Turbinia run complete, build a human-readable message of results.
+    message = 'Completed {0:d} Turbinia tasks\n'.format(len(task_data))
+    for task in task_data:
+      message += '{0!s} ({1!s}): {2!s}\n'.format(
+          task.get('name'),
+          task.get('id'),
+          task.get('status', 'No task status'))
+      for path in task.get('saved_paths') or []:
```
Comment (Onager) [inline dftimewolf/lib/processors/turbinia.py]: task.get('save_paths', [])

File: dftimewolf/lib/processors/turbinia.py [hunk 3]
```diff
@@ -102,26 +102,51 @@ def process(self):
...
+        if path.endswith('worker-log.txt'):
+          continue
+        if path.endswith('{0!s}.log'.format(task.get('id'))):
+          continue
+        if path.startswith('/'):
+          continue
+        message += '  {0:s}\n'.format(path)
+
+    # Store the message for consumption by any reporting modules.
+    self.state.store_data('report', message)
```
Comment (Onager) [inline dftimewolf/lib/processors/turbinia.py]: What if multiple plugins both want to report something? Change to turbinia_report?

File: dftimewolf/lib/processors/turbinia.py (hunk 1 above)
Comment (tomchop) [reply dftimewolf/lib/processors/turbinia.py]: Done.

File: dftimewolf/lib/processors/turbinia.py (hunk 2 above)
Comment (tomchop) [reply dftimewolf/lib/processors/turbinia.py]: Added a comment to explain that `save_paths` may be populated with `None`.

File: dftimewolf/lib/processors/turbinia.py (hunk 3 above)
Comment (tomchop) [reply dftimewolf/lib/processors/turbinia.py]: As discussed, we'll change this with https://github.com/log2timeline/dftimewolf/issues/138

**Pull request merged**